### DISCUSSION TITLE
C# Binding for C++ Gdextension

### DISCUSSION THREAD

**Discussion by GeorgeS2019** [Delsin-Yu/GDExtensionSharp#7] [Ideas]
What will it take to scan and export an extension-api.json of a C++ Gdextension.dll and use the json to create a managed C# wrapper of the exposed properties and methods of the C++ Gdextension, using your current approach?

**Comment by Delsin-Yu** [comment]
I am not sure what's that related to this project.

The purpose of this repo is to discover a way to make coding with GDExtension through C# possible.

**Reply by GeorgeS2019** [reply]
> The purpose of this repo is to discover a way to make coding with **(c++) GDExtension** through C# possible.

Is that what u mean

**Reply by Delsin-Yu** [reply]
Yes, the goal for this project is to allow developers to create unmanaged GDExtension using .Net 8/Native Aot, it could be an alternative to Godot-Mono.
